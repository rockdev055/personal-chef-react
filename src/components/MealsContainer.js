import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchMeals } from  '../redux/modules/Meals/actions'
import Meal from './Meal'

class MealsContainer extends Component {

  componentDidMount() {
    this.props.fetchMeals()
  }
  render() {
    return (
      <div>
        <h1>All Meals</h1>
        {this.props.meals.map(m => <Meal key={m.id} {...m} />)}
      </div>
    )
  }
}

export default connect(state => {
  return {
    meals: state.meals
  }
}, { fetchMeals })(MealsContainer)