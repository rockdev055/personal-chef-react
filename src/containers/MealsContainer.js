import React, { Component } from 'react'
import { connect } from 'react-redux'
import Meal from '../views/Meal'
import { Container } from 'semantic-ui-react'

class MealsContainer extends Component {
  render() {
    return (
      <Container textAlign="center">
        <h1>All Meals</h1>
        {this.props.meals.map(m => <Meal key={m.id} {...m} />)}
      </Container>
    )
  }
}

export default connect(state => {
  return {
    meals: state.meals,
  }
})(MealsContainer)
