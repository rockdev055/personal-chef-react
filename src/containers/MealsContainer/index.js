import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { Link, Route, Switch } from 'react-router-dom';
import { css } from 'glamor';
import NewMeal from '../../components/NewMeal';
import Meal from '../../views/Meal';

const gridContainer = css({
  marginTop: '20px',
  display: 'grid',
  gridTemplateAreas: `"navigation main main"`,
});

const gridNav = css({
  gridArea: 'navigation',
});

const gridMain = css({
  gridArea: 'main',
});

class MealsContainer extends Component {
  render() {
    const { meals } = this.props;
    return (
      <Container>
        <div {...gridContainer}>
          <div {...gridNav}>
            <h1>All Meals</h1>
            <ul>
              {meals.map(m => (
                <Link key={m.id} to={`/meals/${m.id}`}>
                  <h3 style={{ listStyleType: 'none' }}>{m.name}</h3>
                </Link>
              ))}
            </ul>
          </div>
          <div {...gridMain}>
            <Switch>
              <Route path="/meals/new" component={NewMeal} />
              <Route exact path="/meals/:id" render={() => <Meal />} />
            </Switch>
          </div>
        </div>
      </Container>
    );
  }
}

MealsContainer.propTypes = {
  meals: PropTypes.array,
};

export default connect(({ meals }) => ({ meals }))(MealsContainer);
