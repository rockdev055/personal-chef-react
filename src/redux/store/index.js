import {
  combineReducers,
  applyMiddleware,
  createStore
} from 'redux'
import thunk from 'redux-thunk'
import auth from '../modules/Auth/reducer'
import households from '../modules/Households/reducer'
import members from '../modules/Members/reducer'
import meals from '../modules/Meals/reducer'
import { reducer as form } from 'redux-form'

const rootReducer = combineReducers({
  auth,
  form,
  households,
  meals,
  members
})

const middleware = [thunk]

export default createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), applyMiddleware(...middleware))


