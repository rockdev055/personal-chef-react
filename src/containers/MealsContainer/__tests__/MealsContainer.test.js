import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import MealsContainer from '..';
import store from '../../../redux/store';

afterEach(cleanup);

describe('<MealsContainer />', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <MealsContainer />
        </MemoryRouter>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
