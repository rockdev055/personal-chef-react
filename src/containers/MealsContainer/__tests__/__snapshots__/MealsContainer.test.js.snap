// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealsContainer /> matches the snapshot 1`] = `
<div
  data-reactroot=""
  class="ui container"
>
  <div
    data-css-pt6crs=""
  >
    <div
      data-css-4ifvyl=""
    >
      <h1>
        All Meals
      </h1>
      <ul />
    </div>
    <div
      data-css-1jxg4nl=""
    >
      <!-- react-empty: 7 -->
    </div>
  </div>
</div>
`;
